fix: Stop OOM crashes in Execution Data pruning

netroy opened the pull request:
Currently while pruning execution data, we are loading all the data in memory. For instances where there are thousands of executions, this causes the container to run out of memory. Since ids is all we need, we should only query for ids.

[HELP-72](https://linear.app/n8n/issue/HELP-72/modjotothemoon-user-is-unable-to-open-executions-following-instance)

netroy committed:
fix: Stop OOM crashed in Execution Data pruning

Currently while pruning execution data, we are loading all the data in memory. For instances where there are thousands of executions, this causes the container to run out of memory.
Since ids is all we need, we should only query for ids.

janober commented:
Uh wow. Thanks for catching this. But should that not also just happen if we use the not default mode of storing binary data? Because for most people that query should not be required at all.

netroy commented:
> Uh wow. Thanks for catching this. But should that not also just happen if we use the not default mode of storing binary data? Because for most people that query should not be required at all.

This affects only instances where `EXECUTIONS_DATA_PRUNE` is set to `true`. This includes all our cloud customers.

janober commented:
Ah yes, only happens if prune is activated.

But I meant the query does currently only have to run if EXECUTIONS_DATA_PRUNE === true && BINARY_MODE === "filesystem".

For everybody that does not use filesystem mode the query runs and then the data does get discarded again unused.

janober commented:
Or to say it differently. It currently does that query that the binary files on the filesystem get deleted. But for not filesystem mode there are not files to be deleted as the binary data is saved in the database together with the execution.

netroy committed:
query for Executions only when ids are actually needed for pruning binary data

in default mode the binary data is in the database, and will get pruned along with the executions.

janober commented:
Got released with `n8n@0.210.2`
